### ISSUE TITLE
v2.3 breaks MockBaseStrategy

### ISSUE THREAD

**Issue by mkemmerling** [cannatag/ldap3#411]
[Commit 04adf56: fixed regression with newer version of pyasn1 ](https://github.com/cannatag/ldap3/commit/04adf569fecdcbb822783f1b5e1c275285e2373c#diff-0fcbc648a57c341479b5f8a1bf1bec2d) changed line 810 in _ldap3/strategy/mockBase.py_ to check for the existence of the 'initial' key in `node.assertion` before accessing it:

```
-            if node.assertion['initial']:
+            if 'initial' in node.assertion and node.assertion['initial'] is not None:
```
and similarly for the 'any' and 'final' keys in line 815 and 819.

But in line 822 the 'any' and 'final' keys are still accessed withot checking for their existence first:

```
            if substring_filter and not node.assertion['any'] and not node.assertion['final']:  # only initial, adds .*
```
which results in a KeyError if the 'any' or 'final' key is not present.

Changing this to

```
            if substring_filter and not node.assertion.get('any', None) and not node.assertion.get('final', None):  # only initial,adds .*
```
fixes the problem.

**Comment by cannatag** [comment]
Thanks! Please check code in dev.

**Comment by mkemmerling** [comment]
Sorry for coming back to this so late. I simply missed your immediate reply.
The code in dev fixes the problem!

**Comment by cannatag** [comment]
Good. It will be released in version 2.4.

**Comment by mkemmerling** [comment]
Thanks!

**Comment by cannatag** [comment]
Fixed in v2.4
